src/AWS.Deploy.CLI/Commands/CommandFactory.cs

```diff
@@ -226,14 +236,16 @@ private Command BuildDeployCommand()
 
         private Command BuildDeleteCommand()
         {
-            var deleteCommand = new Command("delete-deployment", "Delete an existing deployment.")
+            var deleteCommand = new Command("delete-deployment", "Delete an existing deployment.");
+            lock (s_locker)
             {
-                _optionProfile,
-                _optionRegion,
-                _optionProjectPath,
-                _optionDiagnosticLogging,
-                new Argument("deployment-name")
-            };
+                deleteCommand.AddOption(_optionProfile);
```

Comment:
Why are you using AddOption instead of Add, like the rest of the commands.

Reply:
This is probably done to achieve thread safety. @ganeshnj can weigh more on this.

Reply:
`Add` is overloaded, I agree, we must use `Add` for consistency.

Reply:
Fixed